test/issue-3344/test-spec.k

```diff
@@ -0,0 +1,9 @@
+requires "test.k"
+
+module TEST-SPEC
+    imports TEST
+
+    claim [needed]: <k> run( (SetItem(1) -Set (SetItem(2) REST)) REST) => done(SetItem(1) REST) ... </k>
+        requires notBool (1 in REST)
+
```

Comment:
Can we also add the claim for `requires 1 in REST` too? I believe that was already passing before anyway, but good to codify it as law.

Reply:
Have added the second test.
Unfortunately this integration test has to wait for an update to the K version dependency.
```
env -u KORE_EXEC_OPTS kompile --backend haskell --haskell-backend-command kore-exec -d . test.k
Error:  Compiler: Could not find module: SET-SYMBOLIC
        Source(/build/kore/test/issue-3344/test.k)
        Location(5,5,5,25)
        5 |         imports SET-SYMBOLIC
          .         ^~~~~~~~~~~~~~~~~~~~
make[1]: *** [/build/kore/test/issue-3344/../include.mk:51: test-kompiled/definition.kore] Error 113
```